source/common/tls/utility.cc

```diff
@@ -419,6 +419,18 @@ Envoy::Ssl::ParsedX509NamePtr Utility::parseSubjectFromCertificate(X509& cert) {
   return parseX509NameFromCertificate(cert, CertName::Subject);
 }
 
+uint64_t Utility::getExpirationUnixTime(const X509* cert) {
+  // Obtain the expiration time as system time
+  SystemTime expiration_time = Utility::getExpirationTime(*cert);
+
+  // Convert the time to duration since epoch
+  std::chrono::seconds duration_since_epoch =
+      std::chrono::duration_cast<std::chrono::seconds>(expiration_time.time_since_epoch());
+
+  // Convert the duration to uint64_t and return
+  return static_cast<uint64_t>(duration_since_epoch.count());
+}
```

Comment:
```suggestion
std::chrono::seconds Utility::getExpirationUnixTime(const X509* cert) {
  if (certain == nullptr) {
    return std::chrono::seconds::max();
  }
  // Obtain the expiration time as system time
  SystemTime expiration_time = Utility::getExpirationTime(*cert);

  // Convert the time to duration since epoch
  return std::chrono::duration_cast<std::chrono::seconds>(expiration_time.time_since_epoch());
}
```